src/functions.jl

```diff
@@ -33,12 +29,31 @@ for (mod, funs) in (
     end
 end
 
-# 1 arg before
+# 1 Arg - no default for `dims` keyword
+for (mod, funs) in (
+    (:Base, (:cumsum, :cumprod, :sort, :sort!)),
+)
+    for fun in funs
+        @eval function $mod.$fun(a::NamedDimsArray; dims, kwargs...)
+            numerical_dims = dim(a, dims)
+            data = $mod.$fun(parent(a); dims=numerical_dims, kwargs...)
+            return nameddimsarray_result(a, data, numerical_dims)
+        end
+
+        # Vector case
+        @eval function $mod.$fun(a::NamedDimsArray{L, T, 1}; kwargs...) where {L, T}
+            data = $mod.$fun(parent(a); kwargs...)
+            return NamedDimsArray{NamedDims.names(a)}(data)
+        end
+    end
+end
+
+# 1 arg before - no default for `dims` keyword
 for (mod, funs) in (
     (:Base, (:mapslices,)),
 )
     for fun in funs
-        @eval function $mod.$fun(f, a::NamedDimsArray; dims=:, kwargs...)
+        @eval function $mod.$fun(f, a::NamedDimsArray; dims, kwargs...)
```

Comment:
How does this work without a default for the kwarg `dims`?

Reply:
just the same as Base -- without `dims` specified by the user, it is a `UndefKeywordError`

```julia
julia> A = reshape(1:8, (2, 2, 2));

julia> mapslices(sum, A)
ERROR: UndefKeywordError: keyword argument dims not assigned
Stacktrace:
 [1] mapslices(::Function, ::Array{Int64,3}) at ./abstractarray.jl:1919
 [2] top-level scope at none:0
```